### ISSUE TITLE
SASL_MECHANISM: OAUTHBEARER is not clearly documented

### ISSUE THREAD

**Issue by shane-blacklane** [confluentinc/ksql#6772]
Hi there :wave:

I am trying to delpoy a ksqldb server that will read from a kafka cluster that uses Strimzi for Oauth authentication. I see there is info about how to authenticate ksqldb to a kafka cluster, but only for HTTP-BASIC Authentication/Authorization. The docs fall short on explaining how to use SASL_MECHANISM: OAUTHBEARER.

Even though this problem relates to [this issue](https://github.com/confluentinc/ksql/issues/5254) , and this [PR](https://github.com/confluentinc/ksql/pull/5307/files?short_path=20b075b#diff-20b075b2248e003e16b2ca7866229b89133e8b10087bdb310e585bd69fb1d873),  this section is no longer on the current docs, and it's still not clear to me how to use mechanisms other than http basic auth.

I'm trying to follow the same process used for the kakfka cli. Also this is how we are doing it for [another project](https://gist.github.com/shane-blacklane/b156659e5577614ec8db4934d5693c5e).

I haven't been able to make ksqldb find the strimzi lib/jars, and I'm also not certain that this is the best approach. [This is my Dockerfile](https://gist.github.com/shane-blacklane/b80b9c7897c4a0f521f5d34213751c40)

**Comment by vcrfxia** [comment]
Hi @shane-blacklane , you're asking about configuring SASL authentication between the ksqlDB server and your Kafka cluster, right? Check out these docs here: https://docs.ksqldb.io/en/latest/operate-and-deploy/installation/server-config/security/#configure-ksqldb-for-secured-apache-kafka-clusters

**Comment by shane-blacklane** [comment]
> Hi @shane-blacklane , you're asking about configuring SASL authentication between the ksqlDB server and your Kafka cluster, right? Check out these docs here: https://docs.ksqldb.io/en/latest/operate-and-deploy/installation/server-config/security/#configure-ksqldb-for-secured-apache-kafka-clusters

Hi @vcrfxia thanks for your reply and having a look :smile_cat:. That section of the docs seems to only give an example for `sasl.mechanism=PLAIN` whereas the kafka cluster I want to connect to uses Oauth. Following the `Security Guide` link in the docs you posted I end up [here](https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_oauth.html#configuring-ak-clients), which seems to be suggesting what I am already doing (I think anyway?).

**Comment by vcrfxia** [comment]
Yes, the only concrete example in the ksqlDB docs for configuring SASL authentication between the ksqlDB server and the underlying Kafka cluster is for SASL/PLAIN, but the other SASL mechanisms can be configured in the same way as for other Kafka clients (as you've discovered from the docs you've linked). 

This is good feedback that we should update our docs around this to be more discoverable. cc @JimGalasyn 

Were you able to get your configuration working? If not, more details about the specific errors you're hitting would be helpful for a reproduction attempt. Thanks.

**Comment by shane-blacklane** [comment]
Hi @vcrfxia sorry for the delay in getting back to you on this, I had taken some time away from my computer over the past two weeks :)

I was not able to get the configuration working. The error I'm seeing is:
```
ERROR Failed to start KSQL (io.confluent.ksql.rest.server.KsqlServerMain:66)
org.apache.kafka.common.config.ConfigException: Invalid value io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler for configuration sasl.login.callback.handler.class: Class io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler could not be found.
```
I think all the information about my setup should be in the initial comment but if I've missed anything please let me know

**Comment by vcrfxia** [comment]
Hi @shane-blacklane , I've filed a different Github issue for the issue you've reported, as it's separate from the documentation issue in this one: https://github.com/confluentinc/ksql/issues/6831. Let's track the investigation into the jar not found issue there instead.
